tensorboard/webapp/reloader/reloader_component.ts

```diff
@@ -48,9 +61,23 @@ export class ReloaderComponent {
     });
   }
 
+  private onVisibilityChangeImpl() {
+    if (
+      documentUtil.getVisibilityState() === 'visible' &&
+      this.missedAutoReload
+    ) {
+      this.missedAutoReload = false;
+      this.store.dispatch(reload());
+    }
+  }
+
   private load(reloadPeriodInMs: number) {
     this.reloadTimerId = setTimeout(() => {
-      this.store.dispatch(reload());
+      if (documentUtil.getVisibilityState() === 'visible') {
+        this.store.dispatch(reload());
```

Comment:
Can this make the Reloader reload twice in almost succession? i.e., the other reload is still scheduled to be fired and can dispatch reload event right after this.

Reply:
Yes. I didn't think it would be a big deal but I can easily be convinced otherwise.